tests/components/withings/test_config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Tests for config flow."""
+from aiohttp.test_utils import TestClient
+
+from homeassistant.components.withings import const
+from homeassistant.config import async_process_ha_core_config
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_EXTERNAL_URL,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import DOMAIN as HA_DOMAIN, HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.helpers.config_entry_oauth2_flow import AUTH_CALLBACK_PATH
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_config_non_unique_profile(hass: HomeAssistant) -> None:
+    """Test setup a non-unique profile."""
+    config_entry = MockConfigEntry(
+        domain=const.DOMAIN, data={const.PROFILE: "person0"}, unique_id="0"
+    )
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "profile"}, data={const.PROFILE: "person0"}
+    )
+
+    assert result
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)